File: backend/domain/src/main/kotlin/pt/ira/statistics/StatsOccurrenceImportance.kt
package pt.ira.statistics

import pt.ira.occurrence.OccurrenceType

/**
 * Representa estatísticas agregadas das ocorrências pelo nível de importância.
 *
 * Este modelo encapsula a distribuição de ocorrências conforme o seu nível
 * de prioridade, fornecendo tanto o volume absoluto como a percentagem relativa
 * de cada categoria, permitindo uma análise comparativa da urgência das ocorrências.
 *
 * @property importance Nível de importância das ocorrências (ex: NORMAL, URGENT, CRITICAL).
 * @property count Número absoluto de ocorrências que pertencem a este nível de importância.
 * @property percentage Percentagem relativa deste nível face ao total de ocorrências do sistema.
 *
 * @constructor Cria uma instância de [StatsOccurrenceImportance] com as métricas
 *              agregadas de um nível de importância específico.
 */

data class StatsOccurrenceImportance(
    val importance: OccurrenceType,
    val count: Int,
    val percentage: Double,
)

File: frontend/commons/models/stats/StatsOccurrenceImportance.ts
import { OccurrenceType } from "../occurrence/OccurrenceType";

/**
 * Distribuição de ocorrências por nível de importância.
 * @property importance Nível de importância.
 * @property count Número de ocorrências.
 * @property percentage Percentagem face ao total.
 */
export interface StatsOccurrenceImportance {
  importance: OccurrenceType;
  count: number;
  percentage: number;
}
